scipy/signal/tests/test_signaltools.py

```diff
@@ -3325,16 +3325,16 @@ def test_negative_period_ValueError(self):
 
 
 def cast_tf2sos(b, a):
-    """Convert TF2SOS, casting to complex128 and back to the original dtype."""
+    """Convert TF2SOS, casting to complex64 and back to the original dtype."""
     # tf2sos does not support all of the dtypes that we want to check, e.g.:
     #
-    #     TypeError: array type complex256 is unsupported in linalg
+    #     TypeError: array type complex128 is unsupported in linalg
```

Comment:
I'm surprised by the error message in the test fails. I would've thought that `np.complex128` would be supported by `np.linalg` - it would be the default mode of operation for many calculations (the real and imaginary parts being double size).

Reply:
Yeah, it is in the array API standard too, so I almost wonder if we should be getting a harsher error message for types that basically have to be supported but are falling out as not for some reason.

Reply:
(that said, if this is a legit limitation, wouldn't we get more fails in the suite?) maybe a red herring kind of error message